src/main/proto/service.proto

```diff
@@ -69,3 +71,14 @@ message CreateCustomerRoleRequest {
 message CreateCustomerRoleResponse {
   optional bytes role_id = 1;
 }
+
+message MoveAccountBalanceRequest {
+  optional bytes from_account_id = 1;
+  optional bytes to_account_id = 2;
+  optional string amount = 3;
```

Comment:
Can we add a comment describing the format of amount since it's a free-form string ?

Reply:
This will apply to "balance" of CreateAccountRequest, along with future RPC messages. Should we make some sort of comment at the top or something so as to not repeat this comment throughout the code, or would that be preferred?

Reply:
One option is to use the same format as for spanner numeric types: http://cloud/spanner/docs/reference/rpc/google.spanner.v1 (search for "NUMERIC"):

> Encoded as string, in decimal format or scientific notation format.Decimal format:[+-]Digits[.[Digits]] or[+-][Digits].DigitsScientific notation:[+-]Digits[.[Digits]][ExponentIndicator[+-]Digits] or[+-][Digits].Digits[ExponentIndicator[+-]Digits](ExponentIndicator is "e" or "E")

I haven't checked if this is compatible with `BigDecimal` though.

Anyway probably worth putting a separate short comment on each actually-a-decimal field and then possibly a longer comment at the top about decimals in general.

Reply:
@haseebajmal I followed your example of using the string type (since `createAccount` balance is string). Is there a specific reason why you did string instead of numeric? I was assuming because there's an easy and direct way to convert between string and `BigDecimal`

Reply:
IMO using string (or any other alternatives Chris has suggested) is fine as long as you document well what values are acceptable - precision, +ve/-ve, currency etc., - so that a new user can easily specify the right values.

Repeating the documentation seems fine to me. Usually when browsing through RPC documentation, it's convenient to have this documentation easily accessible at field level. At least I don't usually look at file level comments unless I'm modifying the file.

Reply:
Regd. choice of string in createAccount: Yeah you're right. String to BigDecimal conversion is trivial so I chose that.